dsa-23au/java-dsa/arrays-links/src/main/java/dev/codewithfriends/ArrayWrapper.java

```diff
@@ -44,8 +45,10 @@ public Iterator iterator() {
 
     @Override
     public Object[] toArray() {
-        // Object[] objects = l1.toArray(new String[0]); needs new logic
-        return new Object[0];
+        List<T> al1= new ArrayList<T>();
+        this.add(al1);
+        al1.toArray(); 
```

Comment:
this needs to be changed

Reply:
look forward to compiling.

Reply:
> this needs to be changed

Good idea, will work to add it in the next pull request